### ISSUE TITLE
KeePassXC: Touch ID and Apple Watch unlock support on macOS doesn't work

### ISSUE THREAD

**Issue by dblsaiko** [NixOS/nixpkgs#241103]
### Describe the bug
KeePassXC supports unlocking the database using Touch ID and Apple Watch. This works in the build provided on the upstream GitHub, however in the nixpkgs build, even if enabled it doesn't work.

The build log does consider the features supported:

```
keepassxc> -- Biometry compiler support: TRUE
keepassxc> -- Touch ID compiler support: TRUE
keepassxc> -- Apple watch compiler support: TRUE
```

I haven't looked too much into what would be needed to make this work yet beyond checking the cmake file and build README (I expected detecting these features would fail and it would need additional dependencies, but apparently not).

### Steps To Reproduce
1. Start KeePassXC on macOS
2. Open database
3. Enable Touch ID support in settings
4. Lock database (e.g. by locking the Mac)
5. Unlock database by hitting enter in empty password box

### Expected behavior
It should prompt for unlocking using Touch ID or a prompt on a connected Apple Watch.

### Screenshots
Upstream's build
<img width="772" alt="image" src="https://github.com/NixOS/nixpkgs/assets/3987560/fb84fbeb-39c5-438b-8f09-4e4ea069b5e2">

Nixpkgs build
<img width="914" alt="image" src="https://github.com/NixOS/nixpkgs/assets/3987560/b8d263cf-e169-4c50-b76d-a15e1cdaeaf6">

### Additional context
/

### Notify maintainers

@jonafato @turion @rapenne-s

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.5.0, macOS 13.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.16.1`
 - channels(saiko): `""`
 - channels(root): `""`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

**Comment by 2xsaiko** [comment]
According to @reckenrode this would probably require codesigning to work. (#38624 is the closest thing to a tracking issue for it I guess)

**Comment by fedor-ivn** [comment]
Hey @2xsaiko! Did you manage to resolve this issue in some way? The PR you mentioned above seems to be old and stale.

**Comment by 2xsaiko** [comment]
No, for unlock to work the built app bundle has to be signed, which is not currently possible/implemented because it would have to happen outside of the Nix build. At least I don't think anything has changed in that regard.

**Comment by wq9578** [comment]
Still doesn't work with the Nix package version (only entering a password is offered), while the download version from https://keepassxc.org/download/#macos supports Touch ID flawlessly, once the database is opened and then locked, but not closed.
In both cases it is `KeePassXC - Version 2.7.12`.
